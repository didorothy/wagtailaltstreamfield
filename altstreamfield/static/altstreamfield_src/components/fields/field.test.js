import React from "react";
import { render, cleanup, fireEvent, waitForElement } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Field from "./field";

describe('Field', () => {
    test('#minimal', () => {
        let container = render(<Field css_classes="test" input_id="id_1" label="Test Field" />);
        expect(container.container).toMatchSnapshot();
        cleanup();
    });

    test('#input_css_classes', () => {
        let container = render(<Field css_classes="test" input_css_classes="input field-input" input_id="id_1" label="Test Field" />)
        expect(container.container).toMatchSnapshot();
        cleanup();
    });

    test('#help_text', () => {
        let container = render(<Field css_classes="test" input_id="id_1" label="Test Field" help_text="Sample help text." />)
        expect(container.container).toMatchSnapshot();
        cleanup();
    });

    test('#error', () => {
        let container = render(<Field css_classes="test" input_id="id_1" label="Test Field" help_text="Sample help text." error="This field is required." />)
        expect(container.container).toMatchSnapshot();
        cleanup();
    });
});