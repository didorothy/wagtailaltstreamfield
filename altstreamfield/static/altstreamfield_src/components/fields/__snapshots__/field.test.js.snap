// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field #error 1`] = `
<div>
  <div
    class="test"
  >
    <label
      for="id_1"
    >
      Test Field
    </label>
    <div
      class="field-content"
    >
      <div
        class="input"
      />
      <p
        class="help"
      >
        Sample help text.
      </p>
      <p
        class="error-message"
      >
        This field is required.
      </p>
    </div>
  </div>
</div>
`;

exports[`Field #help_text 1`] = `
<div>
  <div
    class="test"
  >
    <label
      for="id_1"
    >
      Test Field
    </label>
    <div
      class="field-content"
    >
      <div
        class="input"
      />
      <p
        class="help"
      >
        Sample help text.
      </p>
    </div>
  </div>
</div>
`;

exports[`Field #input_css_classes 1`] = `
<div>
  <div
    class="test"
  >
    <label
      for="id_1"
    >
      Test Field
    </label>
    <div
      class="field-content"
    >
      <div
        class="input field-input"
      />
    </div>
  </div>
</div>
`;

exports[`Field #minimal 1`] = `
<div>
  <div
    class="test"
  >
    <label
      for="id_1"
    >
      Test Field
    </label>
    <div
      class="field-content"
    >
      <div
        class="input"
      />
    </div>
  </div>
</div>
`;
